src/components/keyboard-aware-flat-list.ios.js

```diff
@@ -0,0 +1,65 @@
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
+/**
+* @format
+* @flow
+*/
+import React from 'react';
+import { FlatList, Dimensions } from 'react-native';
+
+type PropsType = {
+    ...FlatList.propTypes,
+	shouldPreventAutomaticScroll: void => boolean,
+	parentHeight: number,
+	blockToolbarHeight: number,
+	innerToolbarHeight: number,
+}
+
+const KeyboardAwareFlatList = ( props: PropsType ) => {
+	const {
+		parentHeight,
+		blockToolbarHeight,
+		innerToolbarHeight,
+		shouldPreventAutomaticScroll,
+		...listProps
+	} = props;
+	const { height: fullHeight } = Dimensions.get( 'window' );
+	const keyboardVerticalOffset = fullHeight - parentHeight;
+	const blockHolderPadding = 8;
+	const extraScrollHeight = blockToolbarHeight + innerToolbarHeight + blockHolderPadding + keyboardVerticalOffset;
+
+	return (
+		<KeyboardAwareScrollView
+			style={ { flex: 1 } }
+			keyboardDismissMode={ 'none' }
+			enableResetScrollToCoords={ false }
+			keyboardShouldPersistTaps={ 'handled' }
+			extraScrollHeight={ extraScrollHeight }
+			extraHeight={ 0 }
+			innerRef={ ( ref ) => {
+				( this: any ).flatList = ref;
```

Comment:
I believe that all these `( this: any )` notation in this file are not necessary. I tried changing them to `this.` and it works fine and flow is still happy.

Reply:
Also, this reference seems to be to the `scrollView` and not to the `flatList`? I might be a bit lost 😅

Reply:
> I believe that all these ( this: any ) notation in this file are not necessary. I tried changing them to this. and it works fine and flow is still happy.

flow isn't happy to me :( that was the reason I used `( this: any )` in the first place, maybe you can have error if you run travis check in your local.

<img width="660" alt="screen shot 2018-12-18 at 12 26 02" src="https://user-images.githubusercontent.com/5032900/50144374-1f15a780-02c0-11e9-8250-00e34bac468e.png">


> Also, this reference seems to be to the scrollView and not to the flatList? I might be a bit lost 

right, I was using KeyboardAwareFlatList before and then changed it to KeyboardAwareScrollList but it looks like I forgot updating the name `flatList`. Will push a new commit for this 👍

Reply:
pushed https://github.com/wordpress-mobile/gutenberg-mobile/pull/387/commits/401d69875ec9938f30d20aa1b24df3b6d8188e50

Reply:
> flow isn't happy to me :( that was the reason I used ( this: any ) in the first place, maybe you can have error if you run travis check in your local.

May that be because we are declaring variables "on the fly" in a stateless function component  instead of "properly" declaring them in a class before accessing them?

Reply:
But again, I'm not asking to making that change, just trying to understand the issue 👍

Reply:
yes right.. but we don't have a class here, KeyboardAwareFlatList is a function and we want these variable to be shared among these sub functions of KeyboardAwareFlatList. 
so by using `this` we just insert the variables in the current context, not the ideal solution I agree. 
I tried declaring KeyboardAwareFlatList as a React.PureCompoent/React.Compoent also but it causes unnecessary re-renders because extraData prop of FlatList changes and that change makes KeyboardAwareScrollView re-render. 
Another solution would be declaring these variables in BlockManager and updating them via functions declared in the KeyboardAwareFlatList props, but I think that wouldn't be good since that variables are only needed here in iOS code :(

Reply:
I see now. Sounds good to me! Let's ✅